api/v1/config_types.go

```diff
@@ -232,18 +222,26 @@ type ConfigGitlab struct {
 	// TODO: make every fields optional to reduce duplicate code in ConfigGitlabOverrider
 
 	ProjectID            string `json:"projectID" yaml:"projectID"`
-	Branch               string `json:"branch" yaml:"branch"`
 	PipelineTriggerToken string `json:"pipelineTriggerToken" yaml:"pipelineTriggerToken"`
+	// +optional
+	Branch string `json:"branch,omitempty" yaml:"branch,omitempty"`
+	// InferBranch is for Pull Request's testRunner on gitlab.
+	// If true, samsahai will try to infer the testRunner branch name
+	// from the gitlab MR associated with the PR flow [default: false].
+	// +optional
+	InferBranch bool `json:"inferBranch,omitempty" yaml:"inferBranch,omitempty"`
```

Comment:
should we make it `*bool` instead?
since we should reuse the object with `ConfigGitlabOverrider` in the future